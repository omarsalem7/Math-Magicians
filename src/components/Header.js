import React from 'react';
import { NavLink } from 'react-router-dom';
import './header.css';

const Header = () => {
  const linkStyle = {
    margin: '1rem',
    textDecoration: 'none',
    color: 'orange',
  };
  return (
    <>
      <ul className="header">
        <li>
          <NavLink style={linkStyle} to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink style={linkStyle} to="/calculator">
            Calculator
          </NavLink>
        </li>
        <li>
          <NavLink style={linkStyle} to="/quote">
            Quote
          </NavLink>
        </li>
      </ul>
      <h2 className="title">Math Magicians</h2>
    </>
  );
};

export default Header;
