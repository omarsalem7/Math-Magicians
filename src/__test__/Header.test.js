import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('Test header component', () => {
  it('header renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
