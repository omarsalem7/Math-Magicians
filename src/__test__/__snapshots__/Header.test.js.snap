// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header component header renders correctly 1`] = `
Array [
  <ul
    className="header"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Quote
      </a>
    </li>
  </ul>,
  <h2
    className="title"
  >
    Math Magicians
  </h2>,
]
`;
